packages/test/functional-tests/src/test/ddsHandleEncoding.spec.ts

```diff
@@ -0,0 +1,126 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import {
+	MockFluidDataStoreRuntime,
+	MockStorage,
+	MockDeltaConnection,
+	MockHandle,
+} from "@fluidframework/test-runtime-utils";
+import { DirectoryFactory, IDirectory, MapFactory } from "@fluidframework/map";
+import { IChannel, IChannelFactory } from "@fluidframework/datastore-definitions";
+import { ConsensusRegisterCollectionFactory } from "@fluidframework/register-collection";
+import { detectOutboundReferences } from "@fluidframework/container-runtime";
+
+/**
+ * The purpose of these tests is to demonstrate that DDSes do not do opaque encoding of handles
+ * when preparing the op payload (e.g. prematurely serializing).
+ * This is important because the runtime needs to inspect the full op payload for handles.
+ */
+describe("DDS Handle Encoding", () => {
+	const handle = new MockHandle("whatever");
+	const messages: any[] = [];
+
+	beforeEach(() => {
+		messages.length = 0;
+	});
+
+	/**
+	 * @returns The list of handles found in the given contents object
+	 */
+	function findAllHandles(contents: unknown) {
+		const envelope = { contents, address: "envelope" };
+		const handlesFound: string[] = [];
+		detectOutboundReferences(envelope, (from, to) => {
+			handlesFound.push(to);
+		});
+		return handlesFound;
+	}
+
+	/** A "Mask" over IChannelFactory that specifies the return type of create */
+	interface IChannelFactoryWithCreatedType<T extends IChannel>
+		extends Omit<IChannelFactory, "create"> {
+		create: (...args: Parameters<IChannelFactory["create"]>) => T;
+	}
+
+	/** Each test case runs some code then declares the handles (if any) it expects to be included in the op payload */
+	interface ITestCase {
+		name: string;
+		doStuff(): void;
```

Comment:
`addHandleToContent` or something similar might be a better name because that's the expectation.